apps/meteor/tests/end-to-end/api/chat.ts

```diff
@@ -1460,6 +1467,62 @@ describe('[Chat]', () => {
 					expect(res.body.message).to.have.property('attachments').that.is.an('array').that.has.lengthOf(0);
 				});
 		});
+
+		it('should do nothing if the message text hasnt changed and theres no custom fields', async () => {
+			await request
+				.post(api('chat.update'))
+				.set(credentials)
+				.send({
+					roomId: testChannel._id,
+					msgId: message._id,
+					text: 'This message was edited via API',
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(200)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', true);
+					expect(res.body).to.have.nested.property('message.msg', 'This message was edited via API');
+					expect(res.body).to.not.have.nested.property('message.customFields');
+				});
+		});
+
+		it('should update message custom fields along with msg', async () => {
+			await request
+				.post(api('chat.update'))
+				.set(credentials)
+				.send({
+					roomId: testChannel._id,
+					msgId: message._id,
+					text: 'This message was edited via API 2',
+					customFields: { test: 'test' },
+				})
+				.expect('Content-Type', 'application/json')
+				.expect(200)
+				.expect((res) => {
+					expect(res.body).to.have.property('success', true);
+					expect(res.body).to.have.nested.property('message.msg', 'This message was edited via API 2');
+					expect(res.body.message).to.have.property('customFields').that.is.an('object').that.deep.equals({ test: 'test' });
+				});
+		});
+
+		it('should update message custom fields without changes to msg', async () => {
+			await request
+				.post(api('chat.update'))
+				.set(credentials)
+				.send({
+					roomId: testChannel._id,
+					msgId: message._id,
+					text: 'This message was edited via API 2',
```

Comment:
Can we have a test case for when `text` is `undefined`? Since you made it optional
btw do we really need/want to make it optional?

Reply:
Good question. For me if you can update one without the another, then both should be optional. `msg` can be updated alone as well as `customFields`, so ended up making it optional.

Wonder if that could have any side effects but since making something optional is not breaking we shouldn't.

Gonna add the other test, thanks man!

Reply:
Good one! At the end, the prop couldn't be optional without bigger code changes, so will be kept as required but at least it doesn't have to change now :its_something: